plugins/outputs/opentelemetry/client_http.go

```diff
@@ -0,0 +1,162 @@
+package opentelemetry
+
+import (
+	"bytes"
+	"context"
+	ntls "crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+
+	"github.com/influxdata/telegraf/internal"
+)
+
+type httpClient struct {
+	httpClient   *http.Client
+	url          string
+	encodingType string
+	compress     string
+}
+
+func (h *httpClient) Connect(cfg *clientConfig) error {
+	h.httpClient = &http.Client{}
+	h.url = cfg.ServiceAddress
+	h.encodingType = cfg.Encoding
+	h.compress = cfg.Compression
+
+	if tlsConfig, err := cfg.TLSConfig.TLSConfig(); err != nil {
+		return err
+	} else if tlsConfig != nil {
+		h.httpClient.Transport = &http.Transport{
+			TLSClientConfig: tlsConfig,
+		}
+	} else if cfg.CoralogixConfig != nil {
+		// For coralogix, we enforce HTTP connection with TLS
+		h.httpClient.Transport = &http.Transport{
+			TLSClientConfig: &ntls.Config{},
+		}
+	} else {
+		h.httpClient.Transport = &http.Transport{
+			TLSClientConfig: &ntls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+	return nil
+}
+
+func (h *httpClient) Export(ctx context.Context, request pmetricotlp.ExportRequest) (pmetricotlp.ExportResponse, error) {
+	var err error
+	var requestBytes []byte
+
+	switch h.encodingType {
+	case "application/x-protobuf":
+		requestBytes, err = request.MarshalProto()
+		if err != nil {
+			return pmetricotlp.ExportResponse{}, err
+		}
+	case "application/json":
+		requestBytes, err = request.MarshalJSON()
+		if err != nil {
+			return pmetricotlp.ExportResponse{}, err
+		}
+	default:
+		return pmetricotlp.ExportResponse{}, fmt.Errorf("unsupported content type '%s'", h.encodingType)
+	}
+	var reader io.Reader
+	reader = bytes.NewReader(requestBytes)
+
+	if h.compress != "" && h.compress != "none" {
+		reader = internal.CompressWithGzip(reader)
+	}
+
+	httpRequest, err := http.NewRequestWithContext(ctx, http.MethodPost, h.url, reader)
+	if err != nil {
+		return pmetricotlp.ExportResponse{}, err
+	}
+
+	httpRequest.Header.Set("Content-Type", h.encodingType)
+	httpRequest.Header.Set("User-Agent", userAgent)
+	if h.compress != "" && h.compress != "none" {
+		httpRequest.Header.Set("Content-Encoding", "gzip")
+	}
+
+	httpResponse, err := h.httpClient.Do(httpRequest)
+	if err != nil {
+		return pmetricotlp.ExportResponse{}, err
+	}
+	defer func() {
+		//nolint:errcheck // cannot fail with io.Discard
+		// 64KB is a not specific limit. But, opentelemetry-collector also uses 64KB for safety.
+		io.CopyN(io.Discard, httpResponse.Body, 64*1024)
+		_ = httpResponse.Body.Close()
+	}()
```

Comment:
Why do you need this complexity? Wouldn't

```suggestion
	if _, err := io.Copy(io.Discard, httpResponse.Body) {
		return pmetricotlp.ExportResponse{}, err
	}
```

be good enough?